barman/cloud_providers/google_cloud_storage.py

```diff
@@ -0,0 +1,324 @@
+
+    def test_connectivity(self):
+        """
+        Test gcs connectivity by trying to access a container
+        """
+        try:
+            # We are not even interested in the existence of the bucket,
+            # we just want to see if google cloud storage is reachable.
+            self.bucket_exists = self._check_bucket_existence()
+            return True
+        except GoogleAPIError as exc:
+            logging.error("Can't connect to cloud provider: %s", exc)
+            return False
+
+    def _check_bucket_existence(self):
+        """
+        Check google bucket
+
+        :return: True if the container exists, False otherwise
+        :rtype: bool
+        """
+        return self.container_client.exists()
+
+    def _create_bucket(self):
+        """
+        Create the bucket in cloud storage
+        It will try to create the bucket according to credential provided with 'GOOGLE_APPLICATION_CREDENTIALS' env. This imply the
+        Bucket creation requires following gcsBucket access: 'storage.buckets.create'. Storage Admin role is suited for that.
+
+        It is advised to have the bucket already created. Bucket creation can use a lot of parameters (region, project, dataclass, access control ...).
+        Barman cloud does not provide a way to customise this creation and will use only bucket for creation .
+        You can check detailed documentation here to learn more about default values
+        https://googleapis.dev/python/storage/latest/client.html -> create_bucket
+        """
+        try:
+            self.client.create_bucket(self.container_client)
+        except Conflict as e:
+            logging.error("It seems there was a Conflict creating bucket.")
+            logging.error(e.message)
+            logging.error("The bucket already exist, so we continue.")
+
+    def list_bucket(self, prefix="", delimiter=DEFAULT_DELIMITER):
+        """
+        List bucket content in a directory manner
+
+        :param str prefix: Prefix used to filter blobs
+        :param str delimiter: Delimiter, used with prefix to emulate hierarchy
+        :return: List of objects and dirs right under the prefix
+        :rtype: List[str]
+        """
+        logging.debug("list_bucket: {}, {}".format(prefix, delimiter))
+        blobs = self.client.list_blobs(
+            self.container_client, prefix=prefix, delimiter=delimiter
+        )
+        objects = list(map(lambda blob: blob.name, blobs))
+        dirs = list(blobs.prefixes)
+        logging.debug("objects {}".format(objects))
+        logging.debug("dirs {}".format(dirs))
+        return objects + dirs
+
+    def download_file(self, key, dest_path, decompress):
+        """
+        Download a file from cloud storage
+
+        :param str key: The key identifying the file to download
+        :param str dest_path: Where to put the destination file
+        :param bool decompress: Whenever to decompress this file or not
```

Comment:
`decompress` is either a string or None, not a `bool` - the docstring in `S3CloudInterface` and `CloudInterface`is wrong and the docstring in AzureCloudInterface is correct (I must have failed to update the other two in a previous commit).

Reply:
I will update those docstrings.